source/Mlos.Python/mlos/Optimizers/ExperimentDesigner/ParallelExperimentDesigner.py

```diff
@@ -0,0 +1,215 @@
+#
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT License.
+#
+import pandas as pd
+import numpy as np
+from typing import Dict
+from uuid import UUID, uuid4
+
+from mlos.Logger import create_logger
+from mlos.Optimizers.ExperimentDesigner.ExperimentDesignerBase import ExperimentDesignerBase
+from mlos.Optimizers.RegressionModels.MultiObjectiveRegressionModel import MultiObjectiveRegressionModel
+from mlos.Optimizers.OptimizationProblem import OptimizationProblem
+from mlos.Optimizers.ParetoFrontier import ParetoFrontier
+from mlos.Spaces import CategoricalDimension, ContinuousDimension, Point, SimpleHypergrid
+from mlos.Spaces.Configs.ComponentConfigStore import ComponentConfigStore
+from mlos.Tracer import trace
+
+from .UtilityFunctionOptimizers.RandomSearchOptimizer import RandomSearchOptimizer, random_search_optimizer_config_store
+from .UtilityFunctionOptimizers.GlowWormSwarmOptimizer import GlowWormSwarmOptimizer, glow_worm_swarm_optimizer_config_store
+from .UtilityFunctions.MultiObjectiveProbabilityOfImprovementUtilityFunction import MultiObjectiveProbabilityOfImprovementUtilityFunction,\
+    multi_objective_probability_of_improvement_utility_function_config_store
+
+from .UtilityFunctionOptimizers.UtilityFunctionOptimizerFactory import UtilityFunctionOptimizerFactory
+
+
+parallel_experiment_designer_config_store = ComponentConfigStore(

```

Comment:
Would be nice to reduce code duplication by reusing utility function and numeric numeric optimizers config stores